src/components/ChallengeModal/ChallengeModal.test.tsx

```diff
@@ -0,0 +1,543 @@
+            bot_apikey: "test-key",
+            bot_owner: 1,
+            bot: true,
+            ui_class: "test",
+            settings: {},
+        },
+    ],
+    one_bot: jest.fn(),
+    bot_count: jest.fn(),
+    getAcceptableTimeSetting: jest.fn().mockReturnValue([{ _config_version: 1 }, null]),
+}));
+
+const defaultProps: ChallengeModalProperties = {
+    mode: "player",
+    initialState: null,
+    config: {
+        challenge: {
+            min_ranking: 5,
+            max_ranking: 36,
+            challenger_color: "automatic",
+            invite_only: false,
+            game: {
+                name: "Test Game",
+                rules: "japanese",
+                ranked: false,
+                width: 19,
+                height: 19,
+                handicap: 0,
+                komi_auto: "automatic",
+                disable_analysis: false,
+                initial_state: null,
+                private: false,
+                time_control: {
+                    system: "byoyomi",
+                    speed: "live",
+                    main_time: 1200,
+                    period_time: 30,
+                    periods: 5,
+                    pause_on_weekends: false,
+                },
+            },
+        },
+        conf: {
+            restrict_rank: false,
+            selected_board_size: "19x19",
+        },
+        time_control: {
+            system: "byoyomi",
+            speed: "live",
+            main_time: 1200,
+            period_time: 30,
+            periods: 5,
+            pause_on_weekends: false,
+        },
+    },
+};
+
+const mockModal = {
+    close: jest.fn(),
+    on: jest.fn(),
+    off: jest.fn(),
+};
+
+describe("ChallengeModalBody", () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it("renders game name input", () => {
+        render(<ChallengeModalBody {...defaultProps} modal={mockModal} />);
+        const gameNameInput = screen.getByPlaceholderText("Game Name");
+        expect(gameNameInput).toBeInTheDocument();
+    });
+
+    it("renders private checkbox", () => {
+        render(<ChallengeModalBody {...defaultProps} modal={mockModal} />);
+        const privateCheckbox = screen.getByLabelText("Private");
+        expect(privateCheckbox).toBeInTheDocument();
+    });
+
+    it("renders board size selection", () => {
+        render(<ChallengeModalBody {...defaultProps} modal={mockModal} />);
+        const boardSizeSelect = screen.getByRole("combobox", { name: /board size/i });
+        expect(boardSizeSelect).toBeInTheDocument();
+    });
+
+    it("renders challenger color selection", () => {
+        render(<ChallengeModalBody {...defaultProps} modal={mockModal} />);
+        const colorSelect = screen.getByLabelText(/your color/i);
+        expect(colorSelect).toBeInTheDocument();
+        expect(colorSelect).toHaveValue("automatic");
+    });
+
+    it("renders rank restrictions when enabled", async () => {
+        const openModeProps = {
```

Comment:
Instead of explicitly annotating each property, you can mark the entire object `as const`. This eliminates the need to use `as SomeUnionType` everywhere.  

For example:  

```typescript
const myFoo = {
    someLetter: "a" as LetterType,
    someDigit: 1 as DigitType,
};
```  

can be simplified to:  

```typescript
const myFoo = {
    someLetter: "a",
    someDigit: 1,
} as const;
```  

This approach remains compatible with the type `{ someLetter: LetterType, someDigit: DigitType }`.